packages/react-components/react-table/src/components/TableSelectionCell/useTableSelectionCell.tsx

```diff
@@ -19,27 +19,31 @@ export const useTableSelectionCell_unstable = (
   props: TableSelectionCellProps,
   ref: React.Ref<HTMLElement>,
 ): TableSelectionCellState => {
+  const radioValue = useId('tableradio');
   const tableCellState = useTableCell_unstable(props, ref);
   const { noNativeElements } = useTableContext();
-  const type = props.type ?? 'checkbox';
+  const { type = 'checkbox', checked = false, subtle = false, hidden = false } = props;
 
   return {
     ...tableCellState,
     components: {
       ...tableCellState.components,
       checkboxIndicator: Checkbox,
-      radioIndicator: 'span',
+      radioIndicator: Radio,
     },
     checkboxIndicator: resolveShorthand(props.checkboxIndicator, {
       required: type === 'checkbox',
       defaultProps: { checked: props.checked },
     }),
     radioIndicator: resolveShorthand(props.radioIndicator, {
       required: type === 'radio',
-      defaultProps: { children: <CheckmarkFilled /> },
+      defaultProps: { value: radioValue, name: radioValue },
```

Comment:
In order for the radio to work properly, all radios in the table should share the same `name` (e.g. add it to the table's context). Also, the `value` of the radio would be a way to uniquely identify the radio in the group by something outside of this component, but it's not too useful to set the value to a private generated identifier. If you just want to control whether it's checked, it should work to set the `checked` prop.

```suggestion
      defaultProps: { checked, name: /* from TableContext */ },
```

Reply:
Updated in https://github.com/microsoft/fluentui/pull/25202/commits/5e492610a3372497318be713eb367795dcd98545